_posts/2021-03-04-path-resolution-in-quarkus.adoc

```diff
@@ -0,0 +1,180 @@
+---
+layout: post
+title: Path resolution in Quarkus
+date: 2021-03-04
+tags: help web metrics
+synopsis: Navigating recent changes in Quarkus URI path resolution
+author: ebullient
+---
+
+= Path resolution in Quarkus
+
+We have had an unusually bumpy ride the last few weeks. Path resolution can be sneakily complicated, and in trying to make things better, we accidentally made them worse! We've fixed it all now, but you might notice some changes. Hopefully, this post will make clear what those changes are, what they mean, and what you can do to put everything back the way you want it.
+
+**TL;DR:** Leading slashes matter now in config. So if you use `/endpoint`, that endpoint will be served from the root. If you want it relative to some containing bucket, omit that leading slash. For example, `quarkus.http.non-application-root-path` is now `q` by default, which nests it under `quarkus.http.root-path`, matching the original behavior. You can use an absolute path, e.g. `/q`, to serve non-application endpoints to the absolute root (as a sibling of the http root, if that is set). To remove the non-application endpoint behavior entirely, set `quarkus.http.non-application-root-path` to the same value as `quarkus.http.root-path`. The most foolproof way to do this is using a variable: `quarkus.http.non-application-root-path=${quarkus.http.root-path}`.
```

Comment:
Should "served from the root" in the first sentence say "absolute root" like you do later to prevent confusion with HTTP root?

Reply:
agree!